File: object_initializer.cpp
// includes
#include "object_initializer.h"
#include "generic_protocol/object_initializer.h"
#include "basic_objects/object_initializer.h"

namespace shopndrop_protocol
{

// objects

void initialize( ProductItem * res
    , const std::string &  name
    , const std::string &  unit
    , double price
    , double weight
 )
{
    res->name = name;
    res->unit = unit;
    res->price = price;
    res->weight = weight;
}

void initialize( ShoppingItem * res
    , uint32_t product_item_id
    , uint32_t amount
 )
{
    res->product_item_id = product_item_id;
    res->amount = amount;
}

void initialize( ShoppingList * res
    , const std::vector<ShoppingItem> &  items
 )
{
    res->items = items;
}

void initialize( GeoPosition * res
    , uint32_t plz
    , double latitude
    , double longitude
 )
{
    res->plz = plz;
    res->latitude = latitude;
    res->longitude = longitude;
}

void initialize( RideSummary * res
    , const GeoPosition &  position
    , const basic_objects::LocalTime &  delivery_time
    , double max_weight
 )
{
    res->position = position;
    res->delivery_time = delivery_time;
    res->max_weight = max_weight;
}

void initialize( Ride * res
    , bool is_open
    , const RideSummary &  summary
    , const std::vector<uint32_t> &  pending_order_ids
    , uint32_t accepted_order_id
    , ride_resolution_e resolution
 )
{
    res->is_open = is_open;
    res->summary = summary;
    res->pending_order_ids = pending_order_ids;
    res->accepted_order_id = accepted_order_id;
    res->resolution = resolution;
}

void initialize( Address * res
    , uint32_t plz
    , const std::string &  country
    , const std::string &  city
    , const std::string &  street
    , const std::string &  house_number
    , const std::string &  extra_address_line
 )
{
    res->plz = plz;
    res->country = country;
    res->city = city;
    res->street = street;
    res->house_number = house_number;
    res->extra_address_line = extra_address_line;
}

void initialize( Order * res
    , bool is_open
    , uint32_t ride_id
    , const Address &  delivery_address
    , uint32_t shopping_list_id
    , order_state_e state
    , order_resolution_e resolution
 )
{
    res->is_open = is_open;
    res->ride_id = ride_id;
    res->delivery_address = delivery_address;
    res->shopping_list_id = shopping_list_id;
    res->state = state;
    res->resolution = resolution;
}

// base messages

void initialize( Request * res
    , const std::string &  base_class_param_1
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::Request*>( res ), base_class_param_1 );

}

// messages

void initialize( AddRideRequest * res
    , const std::string &  base_class_param_1
    , const RideSummary &  ride
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->ride = ride;
}

void initialize( AddRideResponse * res
    , uint32_t ride_id
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

    res->ride_id = ride_id;
}

void initialize( CancelRideRequest * res
    , const std::string &  base_class_param_1
    , uint32_t ride_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->ride_id = ride_id;
}

void initialize( CancelRideResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

void initialize( GetRideRequest * res
    , const std::string &  base_class_param_1
    , uint32_t ride_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->ride_id = ride_id;
}

void initialize( GetRideResponse * res
    , const Ride &  ride
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

    res->ride = ride;
}

void initialize( AddOrderRequest * res
    , const std::string &  base_class_param_1
    , uint32_t ride_id
    , const ShoppingList &  shopping_list
    , const Address &  delivery_address
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->ride_id = ride_id;
    res->shopping_list = shopping_list;
    res->delivery_address = delivery_address;
}

void initialize( AddOrderResponse * res
    , uint32_t order_id
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

    res->order_id = order_id;
}

void initialize( CancelOrderRequest * res
    , const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->order_id = order_id;
}

void initialize( CancelOrderResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

void initialize( AcceptOrderRequest * res
    , const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->order_id = order_id;
}

void initialize( AcceptOrderResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

void initialize( DeclineOrderRequest * res
    , const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->order_id = order_id;
}

void initialize( DeclineOrderResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

void initialize( MarkDeliveredOrderRequest * res
    , const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->order_id = order_id;
}

void initialize( MarkDeliveredOrderResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

void initialize( RateShopperRequest * res
    , const std::string &  base_class_param_1
    , uint32_t order_id
    , uint32_t stars
 )
{
    // base class
    initialize( static_cast<Request*>( res ), base_class_param_1 );

    res->order_id = order_id;
    res->stars = stars;
}

void initialize( RateShopperResponse * res
 )
{
    // base class
    ::generic_protocol::initialize( static_cast<generic_protocol::BackwardMessage*>( res ) );

}

// messages (constructors)

AddRideRequest * create_AddRideRequest(
    const std::string &  base_class_param_1
    , const RideSummary &  ride
 )
{
    auto * res = new AddRideRequest;

    initialize( res, base_class_param_1, ride );

    return res;
}

AddRideResponse * create_AddRideResponse(
    uint32_t ride_id
 )
{
    auto * res = new AddRideResponse;

    initialize( res, ride_id );

    return res;
}

CancelRideRequest * create_CancelRideRequest(
    const std::string &  base_class_param_1
    , uint32_t ride_id
 )
{
    auto * res = new CancelRideRequest;

    initialize( res, base_class_param_1, ride_id );

    return res;
}

CancelRideResponse * create_CancelRideResponse(
 )
{
    auto * res = new CancelRideResponse;

    initialize( res );

    return res;
}

GetRideRequest * create_GetRideRequest(
    const std::string &  base_class_param_1
    , uint32_t ride_id
 )
{
    auto * res = new GetRideRequest;

    initialize( res, base_class_param_1, ride_id );

    return res;
}

GetRideResponse * create_GetRideResponse(
    const Ride &  ride
 )
{
    auto * res = new GetRideResponse;

    initialize( res, ride );

    return res;
}

AddOrderRequest * create_AddOrderRequest(
    const std::string &  base_class_param_1
    , uint32_t ride_id
    , const ShoppingList &  shopping_list
    , const Address &  delivery_address
 )
{
    auto * res = new AddOrderRequest;

    initialize( res, base_class_param_1, ride_id, shopping_list, delivery_address );

    return res;
}

AddOrderResponse * create_AddOrderResponse(
    uint32_t order_id
 )
{
    auto * res = new AddOrderResponse;

    initialize( res, order_id );

    return res;
}

CancelOrderRequest * create_CancelOrderRequest(
    const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    auto * res = new CancelOrderRequest;

    initialize( res, base_class_param_1, order_id );

    return res;
}

CancelOrderResponse * create_CancelOrderResponse(
 )
{
    auto * res = new CancelOrderResponse;

    initialize( res );

    return res;
}

AcceptOrderRequest * create_AcceptOrderRequest(
    const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    auto * res = new AcceptOrderRequest;

    initialize( res, base_class_param_1, order_id );

    return res;
}

AcceptOrderResponse * create_AcceptOrderResponse(
 )
{
    auto * res = new AcceptOrderResponse;

    initialize( res );

    return res;
}

DeclineOrderRequest * create_DeclineOrderRequest(
    const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    auto * res = new DeclineOrderRequest;

    initialize( res, base_class_param_1, order_id );

    return res;
}

DeclineOrderResponse * create_DeclineOrderResponse(
 )
{
    auto * res = new DeclineOrderResponse;

    initialize( res );

    return res;
}

MarkDeliveredOrderRequest * create_MarkDeliveredOrderRequest(
    const std::string &  base_class_param_1
    , uint32_t order_id
 )
{
    auto * res = new MarkDeliveredOrderRequest;

    initialize( res, base_class_param_1, order_id );

    return res;
}

MarkDeliveredOrderResponse * create_MarkDeliveredOrderResponse(
 )
{
    auto * res = new MarkDeliveredOrderResponse;

    initialize( res );

    return res;
}

RateShopperRequest * create_RateShopperRequest(
    const std::string &  base_class_param_1
    , uint32_t order_id
    , uint32_t stars
 )
{
    auto * res = new RateShopperRequest;

    initialize( res, base_class_param_1, order_id, stars );

    return res;
}

RateShopperResponse * create_RateShopperResponse(
 )
{
    auto * res = new RateShopperResponse;

    initialize( res );

    return res;
}

} // namespace shopndrop_protocol

File: object_initializer.php
<?php

namespace shopndrop_protocol;


// base include
require_once __DIR__.'/../generic_protocol/object_initializer.php';
// own includes
require_once __DIR__.'/../shopndrop_protocol/protocol.php';

// objects

function initialize__ProductItem( & $res
    , $name // string
    , $unit // string
    , $price // double
    , $weight // double
 )
{
    $res->name = $name;
    $res->unit = $unit;
    $res->price = $price;
    $res->weight = $weight;
}

function initialize__ShoppingItem( & $res
    , $product_item_id // uint32_t
    , $amount // uint32_t
 )
{
    $res->product_item_id = $product_item_id;
    $res->amount = $amount;
}

function initialize__ShoppingList( & $res
    , $items // array<ShoppingItem>
 )
{
    $res->items = $items;
}

function initialize__GeoPosition( & $res
    , $plz // uint32_t
    , $latitude // double
    , $longitude // double
 )
{
    $res->plz = $plz;
    $res->latitude = $latitude;
    $res->longitude = $longitude;
}

function initialize__RideSummary( & $res
    , $position // GeoPosition
    , $delivery_time // basic_objects\LocalTime
    , $max_weight // double
 )
{
    $res->position = $position;
    $res->delivery_time = $delivery_time;
    $res->max_weight = $max_weight;
}

function initialize__Ride( & $res
    , $is_open // bool
    , $summary // RideSummary
    , $pending_order_ids // array<uint32_t>
    , $accepted_order_id // uint32_t
    , $resolution // ride_resolution_e
 )
{
    $res->is_open = $is_open;
    $res->summary = $summary;
    $res->pending_order_ids = $pending_order_ids;
    $res->accepted_order_id = $accepted_order_id;
    $res->resolution = $resolution;
}

function initialize__Address( & $res
    , $plz // uint32_t
    , $country // string
    , $city // string
    , $street // string
    , $house_number // string
    , $extra_address_line // string
 )
{
    $res->plz = $plz;
    $res->country = $country;
    $res->city = $city;
    $res->street = $street;
    $res->house_number = $house_number;
    $res->extra_address_line = $extra_address_line;
}

function initialize__Order( & $res
    , $is_open // bool
    , $ride_id // uint32_t
    , $delivery_address // Address
    , $shopping_list_id // uint32_t
    , $state // order_state_e
    , $resolution // order_resolution_e
 )
{
    $res->is_open = $is_open;
    $res->ride_id = $ride_id;
    $res->delivery_address = $delivery_address;
    $res->shopping_list_id = $shopping_list_id;
    $res->state = $state;
    $res->resolution = $resolution;
}

// base messages

function initialize__Request( & $res
    , $base_class_param_1 // string
 )
{
    // base class
    \generic_protocol\initialize__Request( $res, $base_class_param_1 );

}

// messages

function initialize__AddRideRequest( & $res
    , $base_class_param_1 // string
    , $ride // RideSummary
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->ride = $ride;
}

function initialize__AddRideResponse( & $res
    , $ride_id // uint32_t
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

    $res->ride_id = $ride_id;
}

function initialize__CancelRideRequest( & $res
    , $base_class_param_1 // string
    , $ride_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->ride_id = $ride_id;
}

function initialize__CancelRideResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

function initialize__GetRideRequest( & $res
    , $base_class_param_1 // string
    , $ride_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->ride_id = $ride_id;
}

function initialize__GetRideResponse( & $res
    , $ride // Ride
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

    $res->ride = $ride;
}

function initialize__AddOrderRequest( & $res
    , $base_class_param_1 // string
    , $ride_id // uint32_t
    , $shopping_list // ShoppingList
    , $delivery_address // Address
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->ride_id = $ride_id;
    $res->shopping_list = $shopping_list;
    $res->delivery_address = $delivery_address;
}

function initialize__AddOrderResponse( & $res
    , $order_id // uint32_t
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

    $res->order_id = $order_id;
}

function initialize__CancelOrderRequest( & $res
    , $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->order_id = $order_id;
}

function initialize__CancelOrderResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

function initialize__AcceptOrderRequest( & $res
    , $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->order_id = $order_id;
}

function initialize__AcceptOrderResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

function initialize__DeclineOrderRequest( & $res
    , $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->order_id = $order_id;
}

function initialize__DeclineOrderResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

function initialize__MarkDeliveredOrderRequest( & $res
    , $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->order_id = $order_id;
}

function initialize__MarkDeliveredOrderResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

function initialize__RateShopperRequest( & $res
    , $base_class_param_1 // string
    , $order_id // uint32_t
    , $stars // uint32_t
 )
{
    // base class
    initialize__Request( $res, $base_class_param_1 );

    $res->order_id = $order_id;
    $res->stars = $stars;
}

function initialize__RateShopperResponse( & $res
 )
{
    // base class
    \generic_protocol\initialize__BackwardMessage( $res );

}

// objects (constructors)

function create__ProductItem(
    $name // string
    , $unit // string
    , $price // double
    , $weight // double
 )
{
    $res = new ProductItem;

    initialize__ProductItem( $res, $name, $unit, $price, $weight );

    return $res;
}

function create__ShoppingItem(
    $product_item_id // uint32_t
    , $amount // uint32_t
 )
{
    $res = new ShoppingItem;

    initialize__ShoppingItem( $res, $product_item_id, $amount );

    return $res;
}

function create__ShoppingList(
    $items // array<ShoppingItem>
 )
{
    $res = new ShoppingList;

    initialize__ShoppingList( $res, $items );

    return $res;
}

function create__GeoPosition(
    $plz // uint32_t
    , $latitude // double
    , $longitude // double
 )
{
    $res = new GeoPosition;

    initialize__GeoPosition( $res, $plz, $latitude, $longitude );

    return $res;
}

function create__RideSummary(
    $position // GeoPosition
    , $delivery_time // basic_objects\LocalTime
    , $max_weight // double
 )
{
    $res = new RideSummary;

    initialize__RideSummary( $res, $position, $delivery_time, $max_weight );

    return $res;
}

function create__Ride(
    $is_open // bool
    , $summary // RideSummary
    , $pending_order_ids // array<uint32_t>
    , $accepted_order_id // uint32_t
    , $resolution // ride_resolution_e
 )
{
    $res = new Ride;

    initialize__Ride( $res, $is_open, $summary, $pending_order_ids, $accepted_order_id, $resolution );

    return $res;
}

function create__Address(
    $plz // uint32_t
    , $country // string
    , $city // string
    , $street // string
    , $house_number // string
    , $extra_address_line // string
 )
{
    $res = new Address;

    initialize__Address( $res, $plz, $country, $city, $street, $house_number, $extra_address_line );

    return $res;
}

function create__Order(
    $is_open // bool
    , $ride_id // uint32_t
    , $delivery_address // Address
    , $shopping_list_id // uint32_t
    , $state // order_state_e
    , $resolution // order_resolution_e
 )
{
    $res = new Order;

    initialize__Order( $res, $is_open, $ride_id, $delivery_address, $shopping_list_id, $state, $resolution );

    return $res;
}

// messages (constructors)

function create__AddRideRequest(
    $base_class_param_1 // string
    , $ride // RideSummary
 )
{
    $res = new AddRideRequest;

    initialize__AddRideRequest( $res, $base_class_param_1, $ride );

    return $res;
}

function create__AddRideResponse(
    $ride_id // uint32_t
 )
{
    $res = new AddRideResponse;

    initialize__AddRideResponse( $res, $ride_id );

    return $res;
}

function create__CancelRideRequest(
    $base_class_param_1 // string
    , $ride_id // uint32_t
 )
{
    $res = new CancelRideRequest;

    initialize__CancelRideRequest( $res, $base_class_param_1, $ride_id );

    return $res;
}

function create__CancelRideResponse(
 )
{
    $res = new CancelRideResponse;

    initialize__CancelRideResponse( $res );

    return $res;
}

function create__GetRideRequest(
    $base_class_param_1 // string
    , $ride_id // uint32_t
 )
{
    $res = new GetRideRequest;

    initialize__GetRideRequest( $res, $base_class_param_1, $ride_id );

    return $res;
}

function create__GetRideResponse(
    $ride // Ride
 )
{
    $res = new GetRideResponse;

    initialize__GetRideResponse( $res, $ride );

    return $res;
}

function create__AddOrderRequest(
    $base_class_param_1 // string
    , $ride_id // uint32_t
    , $shopping_list // ShoppingList
    , $delivery_address // Address
 )
{
    $res = new AddOrderRequest;

    initialize__AddOrderRequest( $res, $base_class_param_1, $ride_id, $shopping_list, $delivery_address );

    return $res;
}

function create__AddOrderResponse(
    $order_id // uint32_t
 )
{
    $res = new AddOrderResponse;

    initialize__AddOrderResponse( $res, $order_id );

    return $res;
}

function create__CancelOrderRequest(
    $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    $res = new CancelOrderRequest;

    initialize__CancelOrderRequest( $res, $base_class_param_1, $order_id );

    return $res;
}

function create__CancelOrderResponse(
 )
{
    $res = new CancelOrderResponse;

    initialize__CancelOrderResponse( $res );

    return $res;
}

function create__AcceptOrderRequest(
    $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    $res = new AcceptOrderRequest;

    initialize__AcceptOrderRequest( $res, $base_class_param_1, $order_id );

    return $res;
}

function create__AcceptOrderResponse(
 )
{
    $res = new AcceptOrderResponse;

    initialize__AcceptOrderResponse( $res );

    return $res;
}

function create__DeclineOrderRequest(
    $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    $res = new DeclineOrderRequest;

    initialize__DeclineOrderRequest( $res, $base_class_param_1, $order_id );

    return $res;
}

function create__DeclineOrderResponse(
 )
{
    $res = new DeclineOrderResponse;

    initialize__DeclineOrderResponse( $res );

    return $res;
}

function create__MarkDeliveredOrderRequest(
    $base_class_param_1 // string
    , $order_id // uint32_t
 )
{
    $res = new MarkDeliveredOrderRequest;

    initialize__MarkDeliveredOrderRequest( $res, $base_class_param_1, $order_id );

    return $res;
}

function create__MarkDeliveredOrderResponse(
 )
{
    $res = new MarkDeliveredOrderResponse;

    initialize__MarkDeliveredOrderResponse( $res );

    return $res;
}

function create__RateShopperRequest(
    $base_class_param_1 // string
    , $order_id // uint32_t
    , $stars // uint32_t
 )
{
    $res = new RateShopperRequest;

    initialize__RateShopperRequest( $res, $base_class_param_1, $order_id, $stars );

    return $res;
}

function create__RateShopperResponse(
 )
{
    $res = new RateShopperResponse;

    initialize__RateShopperResponse( $res );

    return $res;
}

// namespace_end shopndrop_protocol


?>
